build.gradle

```diff
@@ -66,6 +66,7 @@ dependencies {
     compile group: 'org.java-websocket', name: 'Java-WebSocket', version: javaWebSocketVersion
     compile group: 'org.slf4j', name: 'slf4j-api', version: slf4jVersion
     compile group: 'org.json', name: 'json', version: jsonOrgVersion
+    compile "org.mockito:mockito-core:2.+"
```

Comment:
I think that the way I wrote it is cleaner and shorter, but whatever you want - your codestyle.
About `testCompile` - not possible since it is used in the main module; I'll explain furthermore in the other comments

Reply:
I understand that the new classes you have added to main requires mockito to also be in main, however, I think those mock classes can be replaced with just mocking them from test with Mockito.